system/ee/ExpressionEngine/Addons/pro_search/libraries/Pro_search_fields.php

```diff
@@ -282,7 +282,10 @@ private function _col($field_id, $col_name, $table = 'grid_columns')
 
         if (! isset($cols[$field_id])) {
             // Init
-            $cols[$field_id] = array();
+            if (!is_array($cols)) {
+                $cols = [];
+                $cols[$field_id] = [];
```

Comment:
Looks like if `$cols` is an array arleady, but `$cols[$field_id]` is not set, then `$cols[$field_id] = []` will not get set. I think if you move the `$cols[$field_id] = [];` assignment after the `if` block it should be good to include.